app/models/custom_attribute.rb

```diff
@@ -1,5 +1,8 @@
 class CustomAttribute < ApplicationRecord
+  ALLOWED_FIELD_TYPES = %w(string integer float boolean date datetime).freeze
   belongs_to :resource, :polymorphic => true
+  default_value_for :field_type, "string"
+  validates :field_type, :inclusion => { :in => ALLOWED_FIELD_TYPES }
```

Comment:
Because of this validation, do you need a data migration to handle any existing records?  Is nil a valid type?